docs/rfcs/0004-suggested-excluded-apps-tech-spec.md

```diff
@@ -0,0 +1,141 @@
+
+Initially, there were two additional goals (no longer considerations):
+-   To have one file for all platforms, so that we'd remember to add a specific app to all platforms at once. With a conversation around potentially wanting different suggested app categories on mobile, this seems unimportant.
+-   Allowing comments in the file. While we still could structure a text file to allow comments, this became less important when the plan became "one file per platform, one app per line".
+
+Other options considered included CSV, YAML, and JSON. While YAML could be useful (as it's easy for less-technical people to read), we don't have a good YAML parser in the app yet. We could add a library, of course. However, we'd need to write our own document format test, because the YAML file (or JSON or CSV) would need to be structured in a specific way.
+
+Ultimately, the multiple text files seemed simpler (and thus easier to build) - there aren't the same parsing concerns and it's easy for contributors.
+
+Metrics
+-------
+
+We will create several metrics as part of this work; they are described in the appropriate spots in the "Implementation" section.
+
+We'd expect the `apps_excluded` metric to increase after this is implemented on platforms which have app exclusions. Both the number of people with more than 0 apps excluded and the average count of apps excluded should increase.
+
+If Nimbus is available by the time this feature is complete, we will use it to roll out the test.
+
+Security / Privacy
+------------------
+
+Since all the work is being done on device, the only security concerns are ensuring we don't log any private info.
+
+The excluded apps list will likely live in the Guardian repo. There are ways to weaponize the suggested apps list to weaken privacy of key apps, but for this attack vector to be used a Mozilla staffer would need to approve the PR with the bad-faith addition to the suggested app list.
+
+Rollout considerations
+----------------------
+
+While the reworked app exclusion page will affect all platforms that support split tunneling, notifications will only be shown to users who have already allowed our app to send them. On mobile, we show the "approve notifications" screen immediately upon launching the app for the first time, which likely depresses approval rates. We may want to re-consider how we show this as part of onboarding.
+
+Test Plan
+---------
+
+-   Fresh install of the app on a device that has a suggested app. Ensure that app is shown appropriately on the screen in VPN Settings. Ensure a notification suggesting its exclusion is popped at an appropriate time.
+-   Install an app on the suggested list on the device. Ensure the app is now shown appropriately on the screen in VPN settings. Ensure a notification suggesting its exclusion is popped at an appropriate time.
+-   Add an app on the device to the list on the repo, wait, and ensure it is offered as an app to exclude within 24 hours.
```

Comment:
It might be helpful for QA to have a more expedient way to test this  - maybe an Inspector Command, or input in the dev menu, by which they can write new apps to the list at will?

Reply:
Good call - I'll add to this doc a dev menu option to immediately run the daily job.